lib/SIL/IR/OperandOwnership.cpp

```diff
@@ -776,14 +777,12 @@ OperandOwnership OperandOwnershipClassifier::visitBuiltinInst(BuiltinInst *bi) {
 //===----------------------------------------------------------------------===//
 
 OperandOwnership Operand::getOperandOwnership() const {
-  // We consider type dependent uses to be instantaneous uses.
-  //
-  // NOTE: We could instead try to exclude type dependent uses from our system,
-  // but that adds a bunch of Optionals and unnecessary types. This doesn't hurt
-  // anything and allows us to eliminate Optionals and thus confusion in between
-  // Optional::None and OwnershipKind::None.
+  // NOTE: NonUse distinguishes itself from InstantaneousUse because it does not
+  // require liveness. Discrimating such uses in the enum avoids the need to
```

Comment:
This can be clearer. My thought:

```
NOTE: NonUse distinguishes itself from InstantaneousUse because it does not require liveness (e.x.: a use of a type dependent operand).
```